jobs/integration/conftest.py

```diff
@@ -301,53 +301,42 @@ async def model(request, tools):
 
 
 @pytest.fixture(scope="module")
-async def k8s_cloud(model, tools):
-    kcp_app = model.applications["kubernetes-control-plane"]
-    kcp_unit = kcp_app.units[0]
-    created_k8s_cloud = False
-
-    with NamedTemporaryFile(dir=Path.home() / ".local" / "share" / "juju") as f:
-        await scp_from(
-            kcp_unit,
-            "config",
-            f.name,
+async def k8s_cloud(kubeconfig, tools):
+    clouds = await tools.run(
+        "juju", "clouds", "--format", "yaml", "-c", tools.controller_name
+    )
+    if tools.k8s_cloud in yaml.safe_load(clouds[0]):
+        yield tools.k8s_cloud
+        return
+
+    _created = False
+    click.echo("Adding k8s cloud")
+    try:
+        await tools.run(
+            "juju",
+            "add-k8s",
+            "--skip-storage",
+            "-c",
             tools.controller_name,
-            tools.connection,
-            proxy=tools.juju_ssh_proxy,
+            tools.k8s_cloud,
         )
```

Comment:
next, let's `add-k8s`.  The `kubeconfig` fixture will handle adding the correct `kubeconfig` file into the `os.environment`